### PR TITLE
Instrumentation now detects Node vs Entry

### PR DESCRIPTION
**Pull request by azy2**

### CODE REVIEW

File: nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashMapShufflingAdder.java [hunk 1]
```diff
@@ -28,6 +28,8 @@ a copy of this software and associated documentation files (the
 
 package edu.illinois.nondex.instr;
 
+import java.util.logging.Level;
```
Comment (alexgyori) [inline nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashMapShufflingAdder.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.logging.Level'](https://www.codacy.com/app/gyori/NonDex/file/2674096265/issues/source?bid=3322087&fileBranchId=3333254#l31)

File: nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashMapShufflingAdder.java [hunk 2]
```diff
@@ -28,6 +28,8 @@ a copy of this software and associated documentation files (the
...
+import java.util.logging.Logger;
```
Comment (alexgyori) [inline nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashMapShufflingAdder.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.logging.Logger'](https://www.codacy.com/app/gyori/NonDex/file/2674096265/issues/source?bid=3322087&fileBranchId=3333254#l32)

File: nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashMapShufflingAdder.java [hunk 3]
```diff
@@ -47,15 +49,23 @@ a copy of this software and associated documentation files (the
     private MethodProperties hasNextProp;
     private MethodProperties nextNodeProp;
 
+    private boolean node;
```
Comment (alexgyori) [inline nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashMapShufflingAdder.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/gyori/NonDex/file/2674096265/issues/source?bid=3322087&fileBranchId=3333254#l52)

File: nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerNodeASMDump.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
+package edu.illinois.nondex.instr;
+import org.objectweb.asm.*;
+public class HashIteratorShufflerNodeASMDump implements Opcodes {
+
+    public static byte[] dump () {
```
Comment (alexgyori) [inline nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerNodeASMDump.java]: Codacy Issue found: [Avoid really long methods.](https://www.codacy.com/app/gyori/NonDex/file/2674096264/issues/source?bid=3322087&fileBranchId=3333254#l5)

File: nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerNodeASMDump.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+        ClassWriter cw = new ClassWriter(0);
+        FieldVisitor fv;
+        MethodVisitor mv;
+        AnnotationVisitor av0;
```
Comment (alexgyori) [inline nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerNodeASMDump.java]: Codacy Issue found: [Avoid unused local variables such as 'av0'.](https://www.codacy.com/app/gyori/NonDex/file/2674096264/issues/source?bid=3322087&fileBranchId=3333254#l10)

File: nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerEntryASMDump.java [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+
+public class HashIteratorShufflerEntryASMDump implements Opcodes {
+
+    public static byte[] dump () {
```
Comment (alexgyori) [inline nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerEntryASMDump.java]: Codacy Issue found: [Avoid really long methods.](https://www.codacy.com/app/gyori/NonDex/file/2674096263/issues/source?bid=3322087&fileBranchId=3333254#l12)

File: nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerEntryASMDump.java [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+        ClassWriter cw = new ClassWriter(0);
+        FieldVisitor fv;
+        MethodVisitor mv;
+        AnnotationVisitor av0;
```
Comment (alexgyori) [inline nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerEntryASMDump.java]: Codacy Issue found: [Avoid unused local variables such as 'av0'.](https://www.codacy.com/app/gyori/NonDex/file/2674096263/issues/source?bid=3322087&fileBranchId=3333254#l17)

**Comment by alexgyori** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           

Clones added
============
- nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerNodeASMDump.java  1
- nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerEntryASMDump.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/gyori/NonDex/pullRequest?prid=249082&bid=3333254)

File: nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerEntryASMDump.java (hunk 7 above)
Comment (alexgyori) [inline nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerEntryASMDump.java]: Codacy Issue found: [Avoid unused local variables such as 'av0'.](https://www.codacy.com/app/gyori/NonDex/file/2674130190/issues/source?bid=3322087&fileBranchId=3333254#l17)

File: nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerEntryASMDump.java (hunk 6 above)
Comment (alexgyori) [inline nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerEntryASMDump.java]: Codacy Issue found: [Avoid really long methods.](https://www.codacy.com/app/gyori/NonDex/file/2674130190/issues/source?bid=3322087&fileBranchId=3333254#l12)

File: nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerNodeASMDump.java (hunk 4 above)
Comment (alexgyori) [inline nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerNodeASMDump.java]: Codacy Issue found: [Avoid really long methods.](https://www.codacy.com/app/gyori/NonDex/file/2674130191/issues/source?bid=3322087&fileBranchId=3333254#l5)

File: nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerNodeASMDump.java (hunk 5 above)
Comment (alexgyori) [inline nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerNodeASMDump.java]: Codacy Issue found: [Avoid unused local variables such as 'av0'.](https://www.codacy.com/app/gyori/NonDex/file/2674130191/issues/source?bid=3322087&fileBranchId=3333254#l10)

**Comment by alexgyori** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Clones added
============
- nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerNodeASMDump.java  1
- nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerEntryASMDump.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/gyori/NonDex/pullRequest?prid=249082&bid=3333254)

**Comment by azy2** [comment]
It's complaining about auto-generated code, I don't know what you want to do about that but I'm inclined to just ignore it.

**Pull request merged**